### ISSUE TITLE
Incompatible with Google Cloud SQL proxy socket connections

### ISSUE THREAD

**Issue by ahmed-sigmalux** [stuttter/ludicrousdb#49]
Google recommends using Google Cloud SQL proxy, which creates a local UNIX socket, to connect to its managed SQL service. These sockets are generated in the format:

`project:region:instance`

So you end up with a local file like:

`/run/cloud_sql_proxy/my-project:us-central1:my-instance`

Using this local file in db-config.php like:

`'host' => 'localhost:/run/cloud_sql_proxy/my-project:us-central1:my-instance`

Generates thousands of global__r errors in the debug.log and basically breaks everything. I think the colons in the local file aren't being escaped / interpreted correctly and that makes the database unreachable.

**Comment by takotakot** [comment]
Does this problem happen WITHOUT this plugin?

**Comment by zardoz-bzar** [comment]
Workaround, make a symlink to their socket.  
  
You should not use Google Cloud SQL for Wordpress. Or ... just about anything. It is a low effort feature made to made AWS on a feature matrix so sales people can say "we have that too".  
  
You are not going to get UDS throughput through a Cloud SQL proxy socket that is actually an TCP/SSL. Using UDS means you have a default concurrency of 128 connections as well, even though it's TCP under the hood.  
  
Don't forget, Google Cloud also treats network traffic as a function of your vcpus, total amount of storage, and aggregates it with disk traffic. Sound confusing? You need 4vcpus and 250GB storage to get 'decent' SQL performance compared to a vanilla 2vpu VPS host.  
  
TL;DR: leave this broken to discourage people from using an awful product from a company that should know better.

Ref: https://github.com/GoogleCloudPlatform/cloudsql-proxy/issues/87

**Comment by JJJ** [comment]
I'm willing to bet (without looking) that the colons are being misinterpreted as ports.
